src/compiler/binder.ts

```diff
@@ -424,6 +425,7 @@ module ts {
                 case SyntaxKind.FunctionDeclaration:
                 case SyntaxKind.FunctionExpression:
                 case SyntaxKind.ArrowFunction:
+                case SyntaxKind.TypeAliasDeclaration:

```

Comment:
Type parameters of classes and interfaces are in the `members` collection, whereas type parameters of generic signatures are in the `locals` collection. Since a type alias doesn't have any notion of members, `locals` seems like the best choice. Also, it is a bit less work in `resolveName` because we don't have to add an extra case to the switch (we look for `locals` on every ancestor node).